recipes/nanopb/all/conanfile.py

```diff
@@ -0,0 +1,46 @@
+from conan import ConanFile
+from conan.tools.cmake import CMake, CMakeDeps, CMakeToolchain, cmake_layout
+from conan.tools.files import apply_conandata_patches, export_conandata_patches, get, copy, rmdir, replace_in_file
+from os import path
+
+class NanoPbConan(ConanFile):
+    name = "nanopb"
+    version = "0.4.8"
+    license = "zlib"
+    url = "https://jpa.kapsi.fi/nanopb/"
+    description = "Protocol Buffers with small code size"
+    settings = "os", "compiler", "build_type", "arch"
+    exports = '*'
+    generators = "CMakeToolchain", "CMakeDeps"
+    options = {
+        "Shared": [True, False], "fPIC": [True, False],
+    }
+    default_options = {
+        "Shared": False, "fPIC": True,
+    }
+
+    tool_requires = "protobuf/3.21.9"
+
+    def layout(self):
+        cmake_layout(self)
+
+    def source(self):
+        get(self, **self.conan_data["sources"][self.version], strip_root=True)
+
+    def configure(self):
+        if self.settings.os == "Windows" and self.settings.compiler == "Visual Studio":
+            del self.options.fPIC
+
+    def build(self):
+        cmake = CMake(self)
+        cmake.configure(source_folder=path.join(self.source_folder, "conan-wrapper"))
+        cmake.build()
+
+    def install(self):
+        cmake = CMake(self)
+        cmake.install()
+
+    def package_info(self):
+        self.cpp_info.includedirs = ["include"]
+        self.cpp_info.libdirs = ["lib"]
+        self.cpp_info.libs = ["protobuf-nanopb"]
```

Comment:
I need to find how to rewrite nanopb_generate_cpp() in the context of using nanopb-config.cmake instead of FindNanopb.cmake.  I found out that in our current build system we use protoc-c to generate the .c/.h file and then we use nanopb_generator.py script on top of those.  So in our current build system we bypass that nanopb_generate_cpp() by using another software.   Will see what I can come up with.  

```
diff --git a/recipes/nanopb/all/conanfile.py b/recipes/nanopb/all/conanfile.py
index 91b22e497..23e96bf6f 100644
--- a/recipes/nanopb/all/conanfile.py
+++ b/recipes/nanopb/all/conanfile.py
@@ -57,13 +57,13 @@ class NanoPbConan(ConanFile):
         self.cpp_info.libs = ["protobuf-nanopb"]
         self.cpp_info.set_property("cmake_find_mode", "none")
         self.cpp_info.builddirs.append(os.path.join("extra"))
+        self.cpp_info.builddirs.append(os.path.join("lib", "cmake"))
 
     def package(self):
         cmake = CMake(self)
         cmake.install()
 
         rmdir(self, os.path.join(self.package_folder, "bin"))
-        rmdir(self, os.path.join(self.package_folder, "lib", "cmake"))
         rmdir(self, os.path.join(self.package_folder, "local"))
 
         copy(self, "*", 
@@ -77,4 +77,3 @@ class NanoPbConan(ConanFile):
         copy(self, "FindNanopb.cmake", 
              os.path.join(self.source_folder, "extra"),
              os.path.join(self.package_folder, "extra"))
-
diff --git a/recipes/nanopb/all/test_package/CMakeLists.txt b/recipes/nanopb/all/test_package/CMakeLists.txt
index 29697d550..57cdcd3e4 100644
--- a/recipes/nanopb/all/test_package/CMakeLists.txt
+++ b/recipes/nanopb/all/test_package/CMakeLists.txt
@@ -2,7 +2,7 @@ cmake_minimum_required(VERSION 3.8)
 
 project(test_package C)
 
-find_package(Nanopb REQUIRED)
+find_package(Nanopb CONFIG REQUIRED)
 
 nanopb_generate_cpp(PROTO_SRC PROTO_HEADER simple.proto)
 ```
 
 if I apply this then I get the correct target in place but loose the nanopb_generate_cpp() macros.